src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -626,10 +626,10 @@ private static bool SymbolIsAnnotatedAsPreview(ISymbol symbol, ConcurrentDiction
                     return true;
                 }
 
-                INamedTypeSymbol? parent = symbol.ContainingType;
-                if (parent is INamespaceSymbol)
+                ISymbol? parent = symbol.ContainingSymbol;
```

Comment:
Just a note for myself. This analyzer uses `ContainingType` everywhere else. Test locally if everything can change to `ContainingSymbol`

Reply:
Fixed now